ffi/capi/Cargo.toml

```diff
@@ -31,11 +31,11 @@ all-features = true
 skip_optional_dependencies = true
 # Bench feature gets tested separately and is only relevant for CI.
 # wearos/freertos/x86tiny are not relevant in normal environments,
-# and smaller_static gets tested on the FFI job anyway
+# smaller_static gets tested on the FFI job anyway
 denylist = ["bench", "wearos", "freertos", "x86tiny", "smaller_static"]
 
 [lib]
-crate-type = ["staticlib", "rlib", "cdylib"]
+crate-type = ["staticlib", "rlib"]
```

Comment:
dylib had to be removed because it doesn't work for emscripten (without what seems to be a large amount of extra effort),

we're not using cdylibs anyway, and it's still _possible_ to compile icu_capi to a cdylib, just not in the emscripten context, and unfortunately https://github.com/rust-lang/cargo/issues/4881 doesn't exist.